feat: enhance `Update-NsxtManagerPasswordComplexity` and `Request-NsxtManagerPasswordComplexity` for `hash_algorithm`

sowjuec opened the pull request:
**Summary of Pull Request**

<!--
    Please provide a clear and concise description of the pull request.
-->
Enhance `Update-NsxtManagerPasswordComplexity` and `Request-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0

Enhance `Update-NsxtManagerPasswordComplexity`  to handle connection to management virtual center

**Type of Pull Request**

<!--
    Please check the one that applies to this pull request using "x".
-->

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

<!--
  Is this related to any GitHub issue(s)?
-->

Closes #47

**Test and Documentation Coverage**

<!--
    Please check the one that applies to this pull request using "x".
-->

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

<!--
    Please check the one that applies to this pull request using "x".
-->

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

<!--
    If this pull request contains a breaking change, please describe the impact and mitigation path.
-->

sowjuec committed:
feat: Enhanced `Update-NsxtManagerPasswordComplexity` and `Request-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0

- Enhanced `Update-NsxtManagerPasswordComplexity` and `Request-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0

-With latest fix of mine, of placing disconnects in the right places, ensured no multiple connections in any method, but  `Update-NsxtManagerPasswordComplexity`  required access to management virtual center as NSX manager is hosted there. Hence ehanced the same.

Ref: #47

Signed-off-by: Sowjanya.V <vsowjanya@vmware.com>

tenthirtyam commented:
Dependent on upstream https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/pull/271.

tenthirtyam commented on CHANGELOG.md:
```diff
@@ -47,10 +47,14 @@ Enhancement:
 - Enhanced `Update-SddcManagerPasswordComplexity` to handle disconnects gracefully. [GH-42](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/42)
 - Enhanced `Get-PasswordPolicyConfig` as there is no significance of default value while parameter is mandatory. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/45)
 - Enhanced `Get-PasswordPolicyDefault` as there is no significance of default value while parameter is mandatory. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/45)
+- Enhanced `Update-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/47)
```
```suggestion
- Enhanced `Update-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for NSX 4.x. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/47)
```

tenthirtyam commented on CHANGELOG.md:
```diff
@@ -47,10 +47,14 @@ Enhancement:
 - Enhanced `Get-PasswordPolicyConfig` as there is no significance of default value while parameter is mandatory. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/45)
 - Enhanced `Get-PasswordPolicyDefault` as there is no significance of default value while parameter is mandatory. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/45)
+- Enhanced `Update-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/47)
+- Enhanced `Request-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for VCF5.0. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/47)
```
```suggestion
- Enhanced `Request-NsxtManagerPasswordComplexity` cmdlet to handle `hash_algorithm` parameter for NSX 4.x. [GH-45](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/pull/47)
```

tenthirtyam reviewed (changes requested):
Can you add the updates to the `./docs/documentation/functions/` files for the cmdlets?

tenthirtyam commented on docs/functions/Update-NsxtManagerPasswordComplexity.md:
````diff
@@ -209,6 +209,22 @@ Default value: 0
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
+### -detail
````
```suggestion
### -hash_algorithm

```

tenthirtyam commented on docs/functions/Update-NsxtManagerPasswordComplexity.md:
```diff
@@ -209,6 +209,22 @@ Default value: 0
+### -detail
+
+The hash/cryptographic algorithm 
+type for new passwords.
```
```suggestion
```

tenthirtyam reviewed (dismissed):
Minor changes to the docs.

sowjuec committed:
Update docs/functions/Update-NsxtManagerPasswordComplexity.md

Signed-off-by: Sowjanya.V <vsowjanya@vmware.com>

Co-authored-by: Ryan Johnson <ryan@tenthirtyam.org>

sowjuec committed:
Update docs/functions/Update-NsxtManagerPasswordComplexity.md

Signed-off-by: Sowjanya.V <vsowjanya@vmware.com>

Co-authored-by: Ryan Johnson <ryan@tenthirtyam.org>

sowjuec committed:
Enhanced ` Update-LocalUserPasswordExpiration` to have connection to management Virtual Center Server

- Enhanced `Update-LocalUserPasswordExpiration` cmdlet to handle connection to management domain vCenter Server instance as NSX Manager virtual machines are placed on the management network.

Ref: #47

Signed-off-by: Sowjanya.V <vsowjanya@vmware.com>

Co-authored-by: Ryan Johnson <ryan@tenthirtyam.org>

Commit:
Update Update-NsxtManagerPasswordComplexity.md

Signed-off-by: Ryan Johnson <johnsonryan@vmware.com>

tenthirtyam commented:
Cherry-picked the commits and opened #49.
